### DISCUSSION TITLE
How do I invoke a cosmwasm contract on a remote chain?

### DISCUSSION THREAD

**Discussion by dckc** [Agoric/agoric-sdk#10414] [Q&A]
@Jovonni says several people have asked this in workshops.

I gather there are ad-hoc solutions. If anybody has one, please share.

Then there's the question of a more convenient API. Brainstorming is welcome.

**Comment by 0xpatrickdev** [comment]
These are presented in the allowlists of some chains' ICA Host Params:

```
/cosmwasm.wasm.v1.MsgInstantiateContract
/cosmwasm.wasm.v1.MsgExecuteContract
```

**Reply by 0xpatrickdev** [reply]
Please follow these issues to stay tuned:

- https://github.com/Agoric/agoric-sdk/issues/10345
- https://github.com/Agoric/agoric-sdk/issues/10353

I can also imagine a yet-to-be-created issue that will call for `.instantiateWasmContract()` and  `.executeWasmContract()` methods directly on an `OrchestrationAccount`.
